File: C#/DesignPatterns/Builder/House.cs
namespace Builder
{
	public record House(IList<Room> Rooms, Roof Roof)
	{
		public void DescribeHouse()
		{
			Console.WriteLine("This is a house with the following rooms:");
			foreach (var room in Rooms)
			{
				Console.WriteLine(room.ToString());
			}

			Console.WriteLine($"And the roof is: {Roof.ToString()}");
		}
	};
}

File: Java/DesignPatterns/src/main/java/Builder/House.java
package Builder;

import java.util.List;

public class House {
    private List<Room> rooms;
    private Roof roof;

    public House(List<Room> rooms, Roof roof) {
        this.rooms = rooms;
        this.roof = roof;
    }

    public void describeHouse() {
        System.out.println("This is a house with the following rooms:");
        for (Room room : rooms) {
            System.out.println(room);
        }
        System.out.println("And the roof is: " + roof);
    }
}
